feature-libs/quote/core/facade/quote.service.ts

```diff
@@ -0,0 +1,490 @@
+  }
+
+  protected requoteCommand: Command<{ quoteStarter: QuoteStarter }, Quote> =
+    this.commandService.create<{ quoteStarter: QuoteStarter }, Quote>(
+      (payload) => {
+        this.isActionPerforming$.next(true);
+        return this.userIdService.takeUserId().pipe(
+          take(1),
+          switchMap((userId) =>
+            this.quoteConnector.createQuote(userId, payload.quoteStarter).pipe(
+              tap((quote) => {
+                this.loadQuoteCartAndProceed(
+                  userId,
+                  quote.cartId as string,
+                  quote.code,
+                  QuoteActionType.REQUOTE
+                );
+              })
+            )
+          )
+        );
+      },
+      {
+        strategy: CommandStrategy.CancelPrevious,
+      }
+    );
+
+  protected quoteDetailsState$: Query<Quote, unknown[]> =
+    this.queryService.create<Quote>(
+      () =>
+        //we need to ensure that the active cart has been loaded, in order to determine if the
+        //quote is connected to a quote cart (and then directly ready for edit)
+        this.activeCartFacade.getActive().pipe(
+          take(1),
+          switchMap(() =>
+            this.routingService.getRouterState().pipe(
+              //we don't need to cover the intermediate router states where a future route is already known.
+              //only changes to the URL are relevant. Otherwise we get unneeded hits when e.g. navigating back from quotes
+              filter((routingData) => routingData.nextState === undefined),
+              withLatestFrom(this.userIdService.takeUserId()),
+              switchMap(([{ state }, userId]) =>
+                this.quoteConnector.getQuote(userId, state.params.quoteId)
+              )
+            )
+          )
+        ),
+
+      {
+        reloadOn: [QuoteDetailsReloadQueryEvent, LoginEvent],
+      }
+    );
+
+  protected getQuotesStateQuery = ({
+    currentPage$,
+    sort$,
+  }: QuotesStateParams) =>
+    this.queryService.create<QuoteList>(
+      () =>
+        this.userIdService.takeUserId().pipe(
+          //use withLatestFrom and reloadOn to get full functionality of query
+          withLatestFrom(currentPage$, sort$),
+          distinctUntilChanged(),
+          switchMap(([userId, currentPage, sort]) => {
+            return this.quoteConnector.getQuotes(userId, {
+              currentPage,
+              sort,
+              pageSize: this.viewConfig.view?.defaultPageSize,
+            });
+          }),
+          tap(() => {
+            this.eventService.dispatch({}, QuoteDetailsReloadQueryEvent);
+          })
+        ),
+      {
+        reloadOn: [
+          uniteLatest([currentPage$, sort$]), // combine all streams that should trigger a reload to decrease initial http calls
+        ],
+        resetOn: [LoginEvent, NavigationEvent],
+      }
+    );
+
+  addDiscount(quoteCode: string, discount: QuoteDiscount): Observable<unknown> {
```

Comment:
Please, add a comment and the following methods as well

Reply:
@ChristophHi, do we need to add comments to the open APIs?

Reply:
Not needed, defined by facade interface